File: src/modules/organization/services/organization_find_one_service.py
from src.modules.organization.models import Organization
from src.shared.types.user_request import UserRequest
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.validators.user_organization_validator import (
    user_organization_validator,
)


class OrganizationFindOneService:
    @classmethod
    def execute(self, id: int, user_request: UserRequest):

        organization = Organization.objects.filter(id=id, deleted_at=None).first()

        if organization is None:
            raise NotFoundException("Organização não encontrada!")

        user_organization_validator(id, user_request)

        return organization

File: src/modules/organization/services/organization-find-one.service.ts
import { prisma } from '../../../infra/database/prisma.service';
import { NotFoundException } from '../../../shared/exceptions';
import { UserRequest } from '../../../shared/types/user-request';
import { userOrganizationValidator } from '../../../shared/validators/user-organization.validator';
import { OrganizationEntity } from '../domain/entities/organization.entity';

class OrganizationFindOneService {
  async execute(
    id: number,
    userRequest: UserRequest,
  ): Promise<OrganizationEntity> {
    const organization = await prisma.organization.findFirst({
      where: {
        id: id,
        deletedAt: null,
      },
      include: {
        address: true,
        avatar: true,
      },
    });

    if (!organization)
      throw new NotFoundException('Organização não encontrada!');

    await userOrganizationValidator(id, userRequest);

    return new OrganizationEntity(organization);
  }
}

export const organizationFindOneService = new OrganizationFindOneService();
